extensions/amp-story/1.0/amp-story.js

```diff
@@ -819,6 +828,23 @@ export class AmpStory extends AMP.BaseElement {
     return storyLayoutPromise;
   }
 
+  /**
+   * Retrieves the initial pageId to begin the story with. In order, the
+   * initial page for a story should be the page ID in the history, a valid
+   * page ID in the URL fragment, and then the first page of the story.
```

Comment:
With your implementation, the page ID in the URL hash will actually override anything else. But I think it's fine, and more aligned with other Google products. For example, on most Google products, a `hl=language` parameter in a URL would override any local preference.

Reply:
+1 to param taking precedence over history state